File: Merge strings/merge_strings.py
def mergeAlternately( word1: str, word2: str) -> str:
    result: str = ""
    
    if len(word1) < len(word2):
        word1 += " " * (len(word2) - len(word1))
    if len(word1) > len(word2):
        word2 += " " * (len(word1) - len(word2))
    
    for i, letter in enumerate(word1):
        result += letter
        result += word2[i]
        
    print(result.replace(" ", ""))
    return result.replace(" ", "")


if __name__ == '__main__':
    assert mergeAlternately(word1="abcd", word2="pq") == "apbqcd"

# Runtime 43 ms Beats 17.61%
# 16.51 MB Beats 74.11%

File: Merge strings/merge_strings.js
function mergeAlternately(word1, word2) {
    let resutl = "";
    const regex = /./;

    if (word1.length > word2.length) {
        word2 = word2.padEnd(word1.length, " ");
    }
    else if (word2.length > word1.length) {
        word1 = word1.padEnd(word2.length, " ");
    }

    for (let i = 0; i < word1.length; i++) {
        resutl += word1[i];
        resutl += word2[i];
    }

    return resutl.replace(/\s/g, "");
};


function test(word1, word2, test) {
    if (mergeAlternately(word1, word2) !== test) {
        throw console.error("");
    }
}

const word1 = "abcd";
const word2 = "pq";
const wordTest = "apbqcd";

console.log(mergeAlternately(word1, word2));
test(word1, word2, wordTest);
